src/app/pagenotfound/pagenotfound.component.ts

```diff
@@ -1,12 +1,15 @@
-import { Component } from '@angular/core';
+import { ServerResponseService } from '../shared/server-response.service';
+import { Component, ChangeDetectionStrategy } from '@angular/core';
 
 @Component({
   selector: 'ds-pagenotfound',
   styleUrls: ['./pagenotfound.component.scss'],
-  templateUrl: './pagenotfound.component.html'
+  templateUrl: './pagenotfound.component.html',
+  providers: [ServerResponseService],
```

Comment:
Is there a particular reason for specifying the provider at the component instead of the module. Doing so will create a separate instance of the provider for each instance of the component. If you specify it at the module level it's a singleton. 

https://angular.io/guide/ngmodule-faq#should-i-add-other-providers-to-a-module-or-a-component

Reply:
No reason, just ignorance :)

The proper place should be in `SharedModule`, isn't it?

Reply:
indeed